seml/__main__.py

```diff
@@ -0,0 +1,678 @@
+        steal_slurm=steal_slurm,
+        no_worker=no_worker,
+        set_to_pending=True,
+        worker_gpus=worker_gpus,
+        worker_cpus=worker_cpus,
+        worker_environment_vars=worker_env,
+    )
+
+
+@app.command("launch-worker")
+def launch_worker_command(
+    ctx: typer.Context,
+    num_exps: NumExperimentsAnnotation = 0,
+    file_output: FileOutputAnnotation = True,
+    steal_slurm: StealSlurmAnnotation = False,
+    post_mortem: PostMortemAnnotation = False,
+    output_to_console: OutputToConsoleAnnotation = False,
+    worker_gpus: WorkerGPUsAnnotation = None,
+    worker_cpus: WorkerCPUsAnnotation = None,
+    worker_env: WorkerEnvAnnotation = None,
+    sacred_id: SacredIdAnnotation = None,
+    filter_dict: FilterDictAnnotation = None,
+    batch_id: BatchIdAnnotation = None,
+):
+    """
+    Launch a local worker that runs PENDING jobs.
+    """
+    assert ctx.obj['collection'], "Collection name must be passed to this command."
+    start_experiments(
+        ctx.obj['collection'],
+        local=True,
+        sacred_id=sacred_id,
+        batch_id=batch_id,
+        filter_dict=filter_dict,
+        num_exps=num_exps,
+        post_mortem=post_mortem,
+        output_to_console=output_to_console,
+        no_file_output=not file_output,
+        steal_slurm=steal_slurm,
+        no_worker=False,
+        set_to_pending=False,
+        worker_gpus=worker_gpus,
+        worker_cpus=worker_cpus,
+        worker_environment_vars=worker_env,
+    )
+
+
+
+@app.command("print-fail-trace")
+def print_traces_command(
+    ctx: typer.Context,
+    sacred_id: SacredIdAnnotation = None,
+    filter_dict: FilterDictAnnotation = None,
+    batch_id: BatchIdAnnotation = None,
+    filter_states: FilterStatesAnnotation = [*States.FAILED, *States.KILLED, *States.INTERRUPTED],
+    yes: YesAnnotation = False,
+):
+    """
+    Prints fail traces of all failed experiments.
+    """
+    assert ctx.obj['collection'], "Collection name must be passed to this command."
+    print_fail_trace(
+        ctx.obj['collection'],
+        sacred_id=sacred_id,
+        filter_states=filter_states,
+        batch_id=batch_id,
+        filter_dict=filter_dict,
+        yes=yes
+    )
+
+
+@app.command("reload-sources")
+def reload_sources_command(
+    ctx: typer.Context,
+    keep_old: Annotated[
+        bool,
+        typer.Option(
+            '-k',
+            '-keep-old',
+            help="Keep the old source files in the database.",
+        ),
+    ] = False,
+    batch_ids: Annotated[list[int], typer.Option(
+        '-b',
+        '--batch-ids',
+        help="Batch IDs (batch_id in the database collection) of the experiments. "
+                "Experiments that were staged together have the same batch_id.",
+    )] = None,
+    yes: YesAnnotation = False,
+):
+    """
+    Reload stashed source files.
+    """
+    assert ctx.obj['collection'], "Collection name must be passed to this command."
```

Comment:
Maybe a minor thing, should we create a function that checks this and outputs a corresponding error message (e.g.  `def assert_collection_specified(specified: bool)`, since this line appears in every command

Reply:
I know added a decorator which enables to flag commands. This is not less code but better code IMO.